contracts/ClubVoting.sol

```diff
@@ -15,11 +15,16 @@ contract Voting {
     mapping(bytes32 => uint256) public votesReceived;
 
     bytes32[] public candidateList;
+    address[] public voterAddresses;
```

Comment:
I agree with you totally on the fact that it is gas intensive! Initially i wanted to just iterate through the map in the contract itself then send the list of voters, but you cant do that in solidity :( 
What if i make a function that just returns the entire map to the caller? will that be gas intensive? because if i do that i could get the entire state at once and would'nt need to call the **voterDetails** function over and over again during auditing

Reply:
Iterating through the map is infeasible since maps cannot be enumerated in solidity due to everything being "Set" by default. 
The array for indexing (like you've implemented) is one possible solution (with an ambiguous amount of gas that will be used - see https://ethereum.stackexchange.com/a/15341)
Other alternatives such as a linked list or double linked list also exist (available at the same link)

Another interesting thing that you can explore (since the dropdown is a nice-to-have feature than a must have) - can be using Events - which are significantly cheaper (see https://ethereum.stackexchange.com/a/73779). If an event is logged on every vote, we can look at all past voters via the `getPastEvents()` call - allowing you to cheaply enumerate the map